pkg/agent/reconcilers.go

```diff
@@ -129,6 +140,27 @@ func (a *Agent) updateWorkStatus(obj runtime.Object) error {
 
 	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
 	defer cancel()
+
+	// delete WorkStatus if exists, when the workload object is deleted
+	if isBeingDeleted {
+		err := a.hubClient.Get(ctx, client.ObjectKeyFromObject(workStatus), workStatus, &client.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+		err = a.hubClient.Delete(ctx, workStatus, &client.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+		a.logger.Info("workStatus deleted", "workStatus-name", workStatus.Name)
+		return nil
+	}
+
```

Comment:
This change is because with the new model ManifestWork can have multiple objects inside and therefore you can no longer rely on deletion of ManifestWork to delete the corresponding WorkStatus, right ?

Reply:
Exactly. And we want to delete a WorkStatus for the object that was removed from the list of objects inside the ManifestWork.